### PR TITLE
Error validation handling formik applicant

### PR DESCRIPTION
**Pull request by melniiv**
## Description :sparkles:
- added translations for different types of validation errors.
- added validation for deminimis form

## Issues :bug:
hds Date input component does not reset the value with resetForm (but is generic, does not work with setting the value with useState).

### CODE REVIEW

File: frontend/benefit/applicant/public/locales/en/common.json [hunk 1]
```diff
@@ -159,5 +159,36 @@
...
+      },
+      "date": {
+        "required": "This field is required",
+        "mustNotInThePast": "This date must not be in the past",
+        "min": "This date must be earliest {{min}}",
+        "max": "This date must be latest {{max}}",
+        "format": "This date is incorrect"
+      },
+      "time": {
+        "required": "This field is required",
```
Comment (Nizar-Rahme) [inline frontend/benefit/applicant/public/locales/en/common.json:188]: `required` seems to be common property, should we move it a step higher?

File: frontend/benefit/applicant/src/components/applications/forms/application/deMinimisAid/DeMinimisAidForm.tsx [hunk 2]
```diff
@@ -61,6 +61,7 @@ const DeMinimisAidForm: React.FC = () => {
               name={fields.deMinimisAidIssueDate.name}
               label={fields.deMinimisAidIssueDate.label}
               placeholder={fields.deMinimisAidIssueDate.placeholder}
+              language="fi"
```
Comment (Nizar-Rahme) [inline frontend/benefit/applicant/src/components/applications/forms/application/deMinimisAid/DeMinimisAidForm.tsx:64]: I prefer if we define an `enum` of the supported languages and use its values instead (e.g: `Lang.FI`).

File: frontend/benefit/applicant/src/components/applications/forms/application/deMinimisAid/extended.ts [hunk 3]
```diff
@@ -83,18 +102,12 @@ const useComponent = (): ExtendedComponentProps => {
...
-    // todo: implement error messages
-    // (getIn(formik.touched, fieldName) || isSubmitted) &&
-    // getIn(formik.errors, fieldName)
-    if (isSubmitted && fieldName) {
-      return fieldName;
-    }
-    return '';
-  };
+  const getErrorMessage = (fieldName: string): string | undefined =>
+    getErrorText(formik.errors, formik.touched, fieldName, t, isSubmitted);
```
Comment (Nizar-Rahme) [inline frontend/benefit/applicant/src/components/applications/forms/application/deMinimisAid/extended.ts:106]: Cannot the return type `string | undefined`  of `getErrorMessage` be inferred?  Does `getErrorText` return `undefined`?

**Review by Nizar-Rahme — CHANGES_REQUESTED** [review]
Left a couple of comments, overall looks good!

File: frontend/benefit/applicant/src/components/applications/forms/application/deMinimisAid/DeMinimisAidForm.tsx (hunk 2 above)
Comment (melniiv) [reply frontend/benefit/applicant/src/components/applications/forms/application/deMinimisAid/DeMinimisAidForm.tsx:64]: yea we can. but it should we always the same value, independent of the locale we choose i guess

Comment (melniiv) [reply frontend/benefit/applicant/src/components/applications/forms/application/deMinimisAid/DeMinimisAidForm.tsx:64]: yes

File: frontend/benefit/applicant/public/locales/en/common.json (hunk 1 above)
Comment (melniiv) [reply frontend/benefit/applicant/public/locales/en/common.json]: yes, it was from another projects, as they have already all translations. probably the idea was that the message for different types might be different. in practice not, so i fixed.

File: frontend/benefit/applicant/src/components/applications/forms/application/deMinimisAid/extended.ts (hunk 3 above)
Comment (melniiv) [reply frontend/benefit/applicant/src/components/applications/forms/application/deMinimisAid/extended.ts:106]: yes, forgot to remove after refactoring, good point.

**Review by Nizar-Rahme — COMMENTED** [review]
LGTM!

**Review by Nizar-Rahme — APPROVED** [review]

**Pull request merged**